include/exadg/structure/time_integration/time_int_gen_alpha.cpp

```diff
@@ -171,10 +171,12 @@ TimeIntGenAlpha<dim, Number>::do_timestep_solve()
   else // linear case
   {
     // solve linear system of equations
-    unsigned int const iter = pde_operator->solve_linear(displacement_np,
-                                                         rhs,
-                                                         this->get_scaling_factor_mass(),
-                                                         this->get_mid_time());
+    unsigned int const iter =
+      pde_operator->solve_linear(displacement_np,
+                                 rhs,
+                                 this->get_scaling_factor_mass(),
+                                 this->get_mid_time(),
+                                 update_preconditioner && (this->time_step_number > 1));
```

Comment:
I would skip `&& (this->time_step_number > 1)` here and do it identically to the nonlinear case. My argument is that the number of updates over a whole unsteady simulation is `N_dt / N_update_interval >> 1`, so the increase in costs is minimal. (Note that the currrent code is potentially problematic because there might come other reasons for preconditioner updates in the future (apart from scaling_factor_mass). Strictly avoiding the preconditioner update in the first call could then cause robustness issues, a bug that might be hard to find once we have forgotton that we have added  `&& (this->time_step_number > 1)` in the present PR.)